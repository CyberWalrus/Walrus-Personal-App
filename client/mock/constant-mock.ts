import { FormType } from "../../config/constants";
import { Option } from "../hocs/with-form-state/with-form-state";
import { InputCustomInterface } from "../type/component";
import { User, UserRole } from "../type/data";

// tslint:disable-next-line:no-empty
const _functVoid = (): void => {};

const FORM_TYPE: FormType = FormType.SIGN_IN;

const user: User = {
  id: `test`,
  email: `test`,
  isActive: true,
  nickName: `test`,
  login: `test`,
  firstName: `test`,
  lastName: `test`,
  userRoleId: [`test`],
  signUpDate: 1234567,
  created: 1234567,
};
const userRole: UserRole = {
  id: `test`,
  name: `test`,
  isActive: true,
};
const optionForm: Option = {
  values: {},
  formErrors: {},
  formValid: true,
  onChangeUserInput: _functVoid,
  onClickSubmit: _functVoid,
};
const optionInput: InputCustomInterface = {
  formTypes: [FormType.SIGN_UP],
  type: `text`,
  name: `login`,
  placeHolder: `Login`,
  hidenValue: `Login`,
};

const usersEmpty: User[] = [];
const users: User[] = [user, user];

const userRolesEmpty: UserRole[] = [];
const userRoles: UserRole[] = [userRole, userRole];

export {
  user,
  usersEmpty,
  users,
  optionForm,
  FORM_TYPE,
  optionInput,
  userRole,
  userRolesEmpty,
  userRoles,
};
