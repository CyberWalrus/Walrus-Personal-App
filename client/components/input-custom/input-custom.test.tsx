import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson, { Json } from "enzyme-to-json";
import * as React from "react";
import { optionForm, optionInput } from "../../mock/constant-mock";
import InputCustom from "./input-custom";

Enzyme.configure({ adapter: new Adapter() });

it(`InputCustom correctly render`, (): void => {
  const tree: Json = toJson(
    shallow(
      <InputCustom
        value={``}
        onChangeInput={optionForm.onChangeUserInput}
        option={optionInput}
      />,
    ),
  );
  expect(tree).toMatchSnapshot();
});
