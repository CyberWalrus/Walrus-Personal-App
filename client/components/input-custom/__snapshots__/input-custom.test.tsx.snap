// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCustom correctly render 1`] = `
<div
  className="form-custom__field"
>
  <input
    className="form-custom__input"
    id="custom-login"
    name="login"
    onChange={[Function]}
    placeholder="Login"
    type="text"
    value=""
  />
  <label
    className="form-custom__label visually-hidden"
    htmlFor="custom-nodejs"
  >
    Login
  </label>
</div>
`;
